src/main/java/uk/gov/justice/digital/delius/ldap/repository/LdapRepository.java

```diff
@@ -81,13 +66,42 @@ public boolean authenticateUser(final String user, final String password) {
                                     NDeliusRole
                                             .builder()
                                             .cn(attributes.get("cn").get().toString())
-                                            .description(attributes.get("description").get().toString())
                                             .build());
 
             return user.toBuilder().roles(roles).build();
         });
     }
 
+    public void addRole(String username, String roleId)  {
```

Comment:
Do you think we need a check for if the user already has this role assigned? LDAP might allow both duplicate roles to exist that could cause issues with removal

Reply:
as discussed - it does a rebind on a DN so should be idempotent.